packages/flutter/lib/src/foundation/debug.dart

```diff
@@ -73,3 +74,21 @@ Future<T> debugInstrumentAction<T>(String description, Future<T> action()) {
 const Map<String, String> timelineWhitelistArguments = <String, String>{
   'mode': 'basic',
 };
+
+/// Configure `debugFormatDouble` using [num.toStringAsPrecision].
+///
+/// Defaults to `null`, which uses the default logic of `debugFormatDouble`.
+int debugDoublePrecision;
+
+/// Formats a double to have standard formatting.
+///
+/// This behavior can be overriden by [debugDoublePrecision].
+String debugFormatDouble(double value) {
```

Comment:
`debug.dart` doesn't really feel like the right place for this since it is not a debug method and called from regular code?

Reply:
Hopefully toDiagnosticsNode and toString are debug code...